app/AvailablePlugin/EmailWidget/Model/EmailWidgetEmail.php

```diff
@@ -0,0 +1,166 @@
+<?php
+/**
+ * COmanage Registry Email Widget Model
+ *
+ * Portions licensed to the University Corporation for Advanced Internet
+ * Development, Inc. ("UCAID") under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * UCAID licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with the
+ * License. You may obtain a copy of the License at:
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * 
+ * @link          http://www.internet2.edu/comanage COmanage Project
+ * @package       registry-plugin
+ * @since         COmanage Registry v4.1.0
+ * @license       Apache License, Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+ */
+
+App::uses("EmailAddress", "Model");
+App::uses("CoMessageTemplate", "Model");
+App::uses("CakeEmail", "Network/Email");
+
+class EmailWidgetEmail extends AppModel {
+  // Define class name for cake
+  public $name = "EmailWidgetEmail";
+
+  // Validation rules for table elements
+  // XXX These are not in use
+  public $validate = array(
+    'email' => 'email',
+    'type' => array(
+      'rule' => 'alphaNumeric',
+      'required' => true
+    ),
+    'token' => array(
+      'rule' => 'alphaNumeric',
+      'required' => true
+    )
+  );
+  
+  /**
+   * Generate a token for email verification and save it in a new record
+   * along with the email address and email type to be  added.
+   * This is step one of a two-step email verification process.
+   * 
+   * @since COmanage Registry v4.1.0
+   * @param string $emailAddress Email address to be added
+   * @param string $emailType    Type of email address to be added 
+   * @return integer id of the new row
+   */
+  public function generateToken($emailAddress, $emailType, $primary) {
+    $token = bin2hex(random_bytes(3));
```

Comment:
This should call `generateRandomToken` (available in `Lib/util.php`). While that function was originally written because `random_bytes` wasn't added until PHP 7 (and at the time we were supporting PHP 5.4), it also skips l and 1 to avoid confusion and inserts dashes into longer tokens.

Reply:
Done.